### PR TITLE
Add CI job to run end-to-end tests

### PR DESCRIPTION
**Pull request by jmsfltchr**
## What is the goal of this PR?

We add CI configuration such that the simulation is run and "end-to-end" tests are run against the resulting knowledge graph.

## What are the changes implemented in this PR?

- Added a new CircleCI job to run the simulation and then run the tests on the resulting data
- Updated the call to tests to include the grakn uri such that when the non-default is needed it is obvious what to change. Also turn off caching, since bazel may cache the test results based on the fact that none of the tests' dependencies have changed, when in fact the code that generates the KB has changed
- Add `.bazelrc`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -26,3 +32,24 @@ jobs:
...
+
+  test-end-to-end:
+    machine: true
+    working_directory: ~/simulation
+    steps:
+      - install-bazel-linux-rbe
+      - checkout
+      - run-grakn-server
+      - run-bazel-rbe:
+          command: bazel run //:simulation-binary -- -k world
```
Comment (haikalpribadi) [inline .circleci/config.yml]: We can rename `:simulation-binary` to just `:simulation`

File: .circleci/config.yml [hunk 2]
```diff
@@ -16,6 +16,12 @@ commands:
           command: bazel run @graknlabs_build_tools//ci:run-bazel-rbe -- << parameters.command >>
           no_output_timeout: 1h
 
+  run-grakn-server:
+    steps:
+      - run: bazel build @graknlabs_grakn_core//:assemble-linux-targz
```
Comment (haikalpribadi) [inline .circleci/config.yml]: We can run this line with RBE, because RBE would have the cache of Grakn build dependencies, and it will simply copy it over to the local machine into `bazel-genfiles/`

File: .circleci/config.yml [hunk 3]
```diff
@@ -26,3 +32,24 @@ jobs:
...
+    machine: true
+    working_directory: ~/simulation
+    steps:
+      - install-bazel-linux-rbe
+      - checkout
+      - run-grakn-server
+      - run-bazel-rbe:
+          command: bazel run //:simulation-binary -- -k world
+      - run-bazel-rbe:
+          command: bazel test //agents/test/... --test_arg=localhost:48555 --cache_test_results=no
```
Comment (haikalpribadi) [inline .circleci/config.yml]: We cannot run this with RBE, because of:
a) the Grakn Server sits locally, and RBE machine is elsewhere without access to the local machine
b) the two test steps are interdependent on each other, and RBE could possibly skip either one if it believes to have been cached

Comment (jmsfltchr) [reply .circleci/config.yml]: Changing to simply use `run` for now, hoping that in the future we can run the tests upon the simulation as a single command with RBE enabled
